### PR TITLE
[codegen/schema] Add a schema checker

### PR DESCRIPTION
**Pull request by pgavlin**
- Change the schema package to report semantic errors as diagnostics
  rather than Go errors
- Add a `pulumi schema check` command to the CLI for static checking of
  package schemas

The semantic checker can be extended in the future to add support for
target-specific checks.

### CODE REVIEW

File: pkg/cmd/pulumi/schema.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/spf13/cobra"
+)
+
+func newSchemaCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "schema",
+		Short: "Analyze package schemas",
+		Long: `Analyze package schemas
+
+Subcommands of this command can be used to analyze Pulumi package schemas. This can be useful to check hand-authored
```
Comment (t0yv0) [inline pkg/cmd/pulumi/schema.go:28]: If this is the correct doc page for what this tool is about, adding a link can be very handy: https://www.pulumi.com/docs/guides/pulumi-packages/schema/

**Review by t0yv0 — APPROVED** [review]
Read through, LGTM, send those diags out.. 🚢

**Review by justinvp — APPROVED** [review]
LGTM

**Review by EvanBoyle — COMMENTED** [review]
Looks great! Would be nice to have a small harness that evaluates a test schema, and checks against an expected set of diagnostic errors. Seems necessary if we are going to expect the providers team (and eventually customers) to depend on the output of this tool.

**Pull request merged**